docs/developer/plugins.md

```diff
@@ -0,0 +1,159 @@
+
+```
+Manifest-Version: 1.0
+Plugin-Class: the.plugin.ClassName
+Plugin-Id: the-plugin-id
+Plugin-Provider: Your Name or Organization
+Plugin-Version: 0.0.0
+```
+
+## Environment variables
+
+`NXF_PLUGINS_MODE`
+: The plugin execution mode, either `prod` for production or `dev` for development (see below for details).
+
+`NXF_PLUGINS_DIR`
+: The path where the plugin archives are loaded and stored (default: `$NXF_HOME/plugins` in production, `./plugins` in development).
+
+`NXF_PLUGINS_DEFAULT`
+: Whether to use the default plugins when no plugins are specified in the Nextflow configuration (default: `true`).
+
+`NXF_PLUGINS_DEV`
+: Comma-separated list of development plugin root directories.
+
+## Development environment
+
+When running in development mode, the plugin system uses the `DevPluginClasspath` to load plugin classes from each plugin project build path, e.g. `plugins/nf-amazon/build/classes` and `plugins/nf-amazon/build/target/libs` (for deps libraries).
+
+## Plugin registry
+
+The metadata for each plugin is published to the [nextflow-io/plugins](https://github.com/nextflow-io/plugins) GitHub repository, specifically [this file](https://raw.githubusercontent.com/nextflow-io/plugins/main/plugins.json).
+
+The repository index has the following structure:
+
+```json
+[
+  {
+    "id": "nf-amazon",
+    "releases": [
+      {
+        "version": "0.2.0",
+        "url": "https://github.com/nextflow-io/nf-amazon/releases/download/0.2.0/nf-amazon-0.2.0.zip",
+        "date": "2020-10-12T10:05:44.28+02:00",
+        "sha512sum": "9e9e33695c1a7c051271..."
+      }
+    ]
+  },
+  /* ... */
+]
+```
+
+## Plugin releases
+
+A plugin release is a ZIP archive containing the plugin classes and the required dependencies.
+
+Nextflow core plugins are released under a corresponding GitHub project, e.g. `nextflow-io/nf-amazon`. This is not a strict requirement, rather it is done to simplify the build process and provide a consistent download experience.
+
+### Installation
+
+Plugins must be declared in the `nextflow.config` file using the `plugins` scope, for example:
+
+```groovy
+plugins {
+    id 'nf-amazon@0.2.0'
+}
+```
+
+If a plugin is not locally available, Nextflow checks the repository index for the download URL, downloads and extracts the plugin archive, and installs the plugin into the directory specified by `NXF_PLUGINS_DIR` (default: `${NXF_HOME}/plugins`).
+
+Since each Nextflow run can have a different set of plugins (and versions), each run keeps a local plugins directory called `.nextflow/plr/<session-id>` which links the exact set of plugins required for the given run.
+
+If no plugins are specified in the Nextflow configuration, the default plugins are automatically added. The set of default plugins is defined in the Nextflow resources file included in the distribution runtime `modules/nextflow/src/main/resources/META-INF/plugins-info.txt`.
```

Comment:
Does this mean that if you include a custom plugin in your config, Nextflow suddenly stops loading all of the default plugins (like amazon etc)? Is that true?

Reply:
No, it always loads the default plugins, I will fix the wording here